NPE in ThingManager on shutdown

[Here](https://community.openhab.org/t/testing-z-wave-binding-on-openhab-2/7522/1383) an NPE has been reported on shutdown:

```
2016-07-10 10:43:14.786 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occured during notification of thing 'zwave:device:364dd67a:node2' about bridge disposal at 'null': null
java.lang.NullPointerException
	at org.eclipse.smarthome.core.thing.internal.ThingManager$11.run(ThingManager.java:835)[102:org.eclipse.smarthome.core.thing:0.9.0.201607061527]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)[:1.7.0_80]
	at java.util.concurrent.FutureTask.run(Unknown Source)[:1.7.0_80]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source)[:1.7.0_80]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)[:1.7.0_80]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)[:1.7.0_80]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)[:1.7.0_80]
	at java.lang.Thread.run(Unknown Source)[:1.7.0_80]
```

There clearly should be a null check and a safer handling within the ThingManager.

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java
@@ -797,12 +797,14 @@ public class ThingManager extends AbstractItemEventSubscriber implements ThingTr
         ThreadPoolManager.getPool(THING_MANAGER_THREADPOOL_NAME).execute(new Runnable() {
             @Override
             public void run() {
+                ThingHandler handler = childThing.getHandler();
                 try {
-                    childThing.getHandler().bridgeHandlerInitialized(bridge.getHandler(), bridge);
+                    if (handler != null) {
+                        handler.bridgeHandlerInitialized(bridge.getHandler(), bridge);
+                    }
                 } catch (Exception ex) {
                     logger.error("Exception occured during notification of thing '" + childThing.getUID()
-                            + "' about bridge initialization at '" + childThing.getHandler() + "': " + ex.getMessage(),
-                            ex);
+                            + "' about bridge initialization at '" + handler + "': " + ex.getMessage(), ex);
                 }
             }
         });
@@ -859,11 +861,14 @@ public class ThingManager extends AbstractItemEventSubscriber implements ThingTr
         ThreadPoolManager.getPool(THING_MANAGER_THREADPOOL_NAME).execute(new Runnable() {
             @Override
             public void run() {
+                ThingHandler handler = childThing.getHandler();
                 try {
-                    childThing.getHandler().bridgeHandlerDisposed(bridge.getHandler(), bridge);
+                    if (handler != null) {
+                        handler.bridgeHandlerDisposed(bridge.getHandler(), bridge);
+                    }
                 } catch (Exception ex) {
                     logger.error("Exception occured during notification of thing '" + childThing.getUID()
-                            + "' about bridge disposal at '" + childThing.getHandler() + "': " + ex.getMessage(), ex);
+                            + "' about bridge disposal at '" + handler + "': " + ex.getMessage(), ex);
                 }
             }
         });

